web-client/src/client/lib.rs

```diff
@@ -526,23 +528,47 @@ impl Client {
                 .subscribed_addresses
                 .borrow()
                 .contains_key(tx.recipient().native_ref());
-        let mut subscribed_address = None;
 
         let consensus = self.inner.consensus_proxy();
 
+        struct AddressSubscription {
+            address: nimiq_keys::Address,
+            consensus: ConsensusProxy,
+        }
+        impl AddressSubscription {
+            pub async fn subscribe(
+                address: nimiq_keys::Address,
+                consensus: ConsensusProxy,
+            ) -> Result<Self, JsError> {
+                consensus
+                    .subscribe_to_addresses(vec![address.clone()], 1, None)
+                    .await?;
+                Ok(Self { address, consensus })
+            }
+        }
+        impl Drop for AddressSubscription {
+            fn drop(&mut self) {
+                let address = self.address.clone();
+                let consensus = self.consensus.clone();
```

Comment:
If these are both options you can `.take()` instead of `.clone()`. After they are moved the struct is droped and the `None` vanish, so there is no real need for the clones. Sadly as far as i know there is otherwise only `unsafe` ways of taking ownership here.

Reply:
So is your suggestion to make them options?

Reply:
Yes I would do that. It is a very small change and makes it more clean in my opinion.